src/Common/src/Interop/Windows/secur32/ApplicationProtocols.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System.Net
+{
+    // _SEC_APPLICATION_PROTOCOLS
+    [StructLayout(LayoutKind.Sequential,Pack = 1)]
+    internal struct ApplicationProtocols
+    {
+        private static readonly int ProtocolListOffset = Marshal.SizeOf<ApplicationProtocols>();
+        private static readonly int ProtocolListConstSize = ProtocolListOffset - (int)Marshal.OffsetOf<ApplicationProtocols>("ProtocolExtenstionType");
+        public uint ProtocolListsSize;
+        public ApplicationProtocolNegotiationExt ProtocolExtenstionType;
+        public short ProtocolListSize;
+
+        public static unsafe byte[] ToByteArray(string[] applicationProtocols)
+        {
+            long protocolListSize = 0;
+            foreach (string applicationProtocol in applicationProtocols)
+            {
+                if (string.IsNullOrEmpty(applicationProtocol)) throw new ArgumentException(SR.net_ssl_app_protocols_invalid, "applicationProtocols");
+                if (applicationProtocol.Length > byte.MaxValue) throw new ArgumentException(SR.net_ssl_app_protocols_invalid, "applicationProtocols");       
+                protocolListSize += applicationProtocol.Length + 1;
+                if (protocolListSize > short.MaxValue)
+                    throw new ArgumentException(SR.net_ssl_app_protocols_invalid, "applicationProtocols");
+            }
+
+            ApplicationProtocols protocols = new ApplicationProtocols();
+            protocols.ProtocolListsSize = (uint)(ProtocolListConstSize + protocolListSize);
+            protocols.ProtocolExtenstionType = ApplicationProtocolNegotiationExt.ALPN;
+            protocols.ProtocolListSize = (short)protocolListSize;
+
+            byte[] buffer = new byte[ProtocolListOffset + protocolListSize];
+            fixed (byte* bufferPtr = buffer)
+            {
+                Marshal.StructureToPtr(protocols, new IntPtr(bufferPtr), false);
```

Comment:
Is a corresponding DeleteStructure call needed in order to clean up?  From a cursory look, it looks like the answer is "no" due to the shape of ApplicationProtocols, but it'd be good to verify and then either fix or add a comment explaining why it's not needed.  Otherwise it's a possible leak.